meerk40t/gui/scene/guicolors.py

```diff
@@ -20,58 +20,59 @@ class GuiColors():
     '''
     def __init__(self, context):
         self.context = context
-        self.default_color = (
-            wx.Colour(0xA0, 0x7F, 0xA0),
-            wx.Colour(0x00, 0x00, 0xFF, 0x40),
-            wx.Colour(0xA0, 0xA0, 0xA0, 128),
-            wx.Colour(0x00, 0x00, 0x00),
-            wx.Colour("Grey"),
-            wx.Colour(0xFF, 0xA0, 0xA0, 0x60),
-            wx.Colour(0xA0, 0xA0, 0xA0, 0x40),
-            wx.Colour(0x00, 0xFF, 0x00, 0xA0),
-            wx.Colour(0xFF, 0x00, 0x00),
-            wx.Colour(0x00, 0xFF, 0x00),
-            wx.Colour(0x00, 0x00, 0xFF),
-            wx.Colour(0xFF, 0xFF, 0xFF)
-        )
-
-        self.context.setting(str, "color_manipulation", color_to_str(self.default_color[0].GetRGBA()))
-        self.context.setting(str, "color_laserpath", color_to_str(self.default_color[1].GetRGBA()))
-        self.context.setting(str, "color_grid", color_to_str(self.default_color[2].GetRGBA()))
-        self.context.setting(str, "color_guide", color_to_str(self.default_color[3].GetRGBA()))
-        self.context.setting(str, "color_background", color_to_str(self.default_color[4].GetRGBA()))
-        self.context.setting(str, "color_magnetline", color_to_str(self.default_color[5].GetRGBA()))
-        self.context.setting(str, "color_snap_visible", color_to_str(self.default_color[6].GetRGBA()))
-        self.context.setting(str, "color_snap_closeup", color_to_str(self.default_color[7].GetRGBA()))
-        self.context.setting(str, "color_selection1", color_to_str(self.default_color[8].GetRGBA()))
-        self.context.setting(str, "color_selection2", color_to_str(self.default_color[9].GetRGBA()))
-        self.context.setting(str, "color_selection3", color_to_str(self.default_color[10].GetRGBA()))
-        self.context.setting(str, "color_bed", color_to_str(self.default_color[11].GetRGBA()))
+        self.default_color = {
+            "manipulation": wx.Colour(0xA0, 0x7F, 0xA0),
+            "laserpath": wx.Colour(0x00, 0x00, 0xFF, 0x40),
+            "grid": wx.Colour(0xA0, 0xA0, 0xA0, 128),
+            "guide": wx.Colour(0x00, 0x00, 0x00),
+            "background": wx.Colour("Grey"),
+            "magnetline": wx.Colour(0xFF, 0xA0, 0xA0, 0x60),
+            "snap_visible": wx.Colour(0xA0, 0xA0, 0xA0, 0x40),
+            "snap_closeup": wx.Colour(0x00, 0xFF, 0x00, 0xA0),
+            "selection1": wx.Colour(0xFF, 0x00, 0x00),
+            "selection2": wx.Colour(0x00, 0xFF, 0x00),
+            "selection3": wx.Colour(0x00, 0x00, 0xFF),
+            "bed": wx.Colour(0xFF, 0xFF, 0xFF)
+        }
+
+        self.context.setting(str, "color_manipulation", color_to_str(self.default_color["manipulation"].GetRGBA()))
```

Comment:
Part of the reason for the default color dict is so you can do a for loop here.
```python
for key in self.default_color:
    self.context.setting(str, f"color_{key}", color_to_str(self.default_color[key].GetRGBA()))
```

The goal being that because we have a dictionary we can merely add things to the dictionary and all our code still works. Minimum effort to add colors to this code.

Reply:
I'll do that in the next release when a GUI to change the colors will be made available.